File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int tar) {
        int low = 0, high = nums.size()-1;
        while(low<high){
            if( nums[low]+nums[high]==tar) return {low+1,high+1};
            else if (nums[low]+nums[high] > tar) --high;
            else if (nums[low]+nums[high]<tar) ++low;
        }
        return {};
    }
};

File: Data Structures & Algorithms/is-palindrome/submission-2.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        clean_s = "".join(_.lower() for _ in s if _.isalnum())
        low, high = 0, len(clean_s)-1
        while low<high:
            if clean_s[low]!=clean_s[high]:
                return False
            low +=1
            high -=1
        return True